NitroxClient/MonoBehaviours/BaseLeakManager.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections.Generic;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Packets;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours;
+
+public class BaseLeakManager : MonoBehaviour
+{
+    private Dictionary<Int3, NitroxId> idByRelativeCell;
+    private Base @base;
+    private BaseHullStrength baseHullStrength;
+    private NitroxId baseId;
+
+    public void Awake()
+    {
+        if (!TryGetComponent(out baseHullStrength))
+        {
+            Log.Error($"Tried adding a {nameof(BaseLeakManager)} to a GameObject that isn't a base, deleting it.");
+            Destroy(this);
+            return;
+        }
+        @base = baseHullStrength.baseComp;
+        @base.TryGetNitroxId(out baseId);
+        idByRelativeCell = new();
+    }
+
+    /// <summary>
+    /// Either creates or updates existing leaks by modifying the base cell's health.
+    /// Also registers the leak's id for further use.
+    /// </summary>
+    public void EnsureLeak(Int3 relativeCell, NitroxId cellId, float health)
+    {
+        Int3 absoluteCell = Absolute(relativeCell);
+        Transform cellObject = @base.GetCellObject(absoluteCell);
+        if (!cellObject)
+        {
+            return;
+        }
+
+        idByRelativeCell[relativeCell] = cellId;
+
+        if (cellObject.TryGetComponent(out LiveMixin cellLiveMixin))
+        {
+            float deltaHealth = health - cellLiveMixin.health;
+            if (Mathf.Abs(deltaHealth) > 1)
+            {
+                // Useful part of BaseHullStrength.CrushDamageUpdate
+                this.Resolve<LiveMixinManager>().SyncRemoteHealth(cellLiveMixin, health, cellObject.position, DamageType.Pressure);
+                // Only play noise if the leak lost health
+                if (deltaHealth < 0)
+                {
+                    // Spawning multiple leaks would result in a big sounds when loading the game
+                    if (Multiplayer.Main && Multiplayer.Main.InitialSyncCompleted)
+                    {
+                        int num = 0;
+                        if (baseHullStrength.totalStrength <= -3f)
+                        {
+                            num = 2;
+                        }
+                        else if (baseHullStrength.totalStrength <= -2f)
+                        {
+                            num = 1;
+                        }
+                        if (baseHullStrength.crushSounds[num] != null)
+                        {
+                            Utils.PlayFMODAsset(baseHullStrength.crushSounds[num], cellObject, 20f);
+                        }
+                    }
+                    ErrorMessage.AddMessage(Language.main.GetFormat("BaseHullStrDamageDetected", baseHullStrength.totalStrength));
```

Comment:
This would make a wall of text during initial sync. I'll move this line in the condition above so it only happens after initial sync and copy it to the base entity loading so that we still have at least one of those messages.